### ISSUE TITLE
Server crashes when trying to open connection to add device -or- No screen info.

### ISSUE THREAD

**Issue by mbarnes-arrdude** [openstf/stf#1152]
I have a weird device. It is old and unsupported by the NDK. It uses a mips processor (I can hear the groans from here).

```
$ adb devices -l
List of devices attached
Ingenic-99431717-1fc06803-f91c0000-886019af device usb:337641472X transport_id:3
```

```
[ro.product.board]: [dwin]
[ro.product.brand]: [Ingenic]
[ro.product.cpu.abi2]: [armeabi-v7a]
[ro.product.cpu.abi]: [mips]
[ro.product.device]: [dwin]
[ro.product.locale.language]: [zh]
[ro.product.locale.region]: [CN]
[ro.product.manufacturer]: [Ingenic]
[ro.product.model]: [SmartTab1]
[ro.product.name]: [dwin]
```
Out of the box STF tries to load to the device the minicap exe and lib for the alternate abi `armeabi--v7a` but it fails to load the prebuilt:

```
2020-01-01T23:11:13.860Z INF/device:plugins:display 7033 [Ingenic-99431717-1fc06803-f91c0000-886019af] Reading display info
2020-01-01T23:11:14.486Z FTL/device 7033 [Ingenic-99431717-1fc06803-f91c0000-886019af] Setup had an error Error: link_image[2222]:  1456 could not load needed library 'minicap.so' for '/data/local/tmp/minicap' (load_arm_library[1235]: Load ARM minicap.so failed!)CANNOT LINK EXECUTABLE

    at /Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/lib/units/device/plugins/util/display.js:43:19
    at tryCatcher (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/promise.js:503:31)
    at Promise._settlePromiseAt (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/promise.js:577:18)
    at Async._drainQueue (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/async.js:128:12)
    at Async._drainQueues (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/adbkit/node_modules/bluebird/js/main/async.js:15:14)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
```

I have hacked a custom build of libjpeg-turbo and minicap for mips by naively adding the arch to the Application.mk files and adding the appropriate targets.

Testing minicap with autosize "seems" to work from the command line. To use it, I copied the bin and lib files into the prebuilt directory in source. In doing this I created the required directory structure. It seems to work great. STF recognizes the device, the arch, and the android version and injects my custom minicap and lib.

STF itself crashes sometimes at this point. If it does not crash, it is unable to show the screencap in the UI. Control buttons for orientation and display on/off work so the agent has some functionality.

When STF crashes it gives me this error:

```
2020-01-01T22:30:04.818Z INF/provider 6953 [*] Restarting device worker "Ingenic-99431717-1fc06803-f91c0000-886019af"
2020-01-01T22:30:05.602Z INF/device:support:push 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Sending output to "tcp://127.0.0.1:7116"
2020-01-01T22:30:05.604Z INF/device 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Preparing device
2020-01-01T22:30:05.896Z INF/device:support:sub 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Receiving input from "tcp://127.0.0.1:7114"
2020-01-01T22:30:05.897Z INF/device:support:sub 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Subscribing to permanent channel "*ALL"
2020-01-01T22:30:05.932Z INF/device:support:properties 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Loading properties
2020-01-01T22:30:05.967Z INF/device:support:sdk 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Supports SDK 16
2020-01-01T22:30:05.968Z INF/device:support:abi 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Supports ABIs mips, armeabi-v7a
2020-01-01T22:30:06.029Z INF/device:resources:minicap 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Installing "/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/minicap-prebuilt/prebuilt/mips/bin/minicap" as "/data/local/tmp/minicap"
2020-01-01T22:30:06.029Z INF/device:resources:minicap 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Installing "/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/minicap-prebuilt/prebuilt/mips/lib/android-16/minicap.so" as "/data/local/tmp/minicap.so"
2020-01-01T22:30:07.758Z INF/device:resources:service 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Checking whether we need to install STFService
2020-01-01T22:30:08.447Z INF/device:resources:service 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Running version check
2020-01-01T22:30:09.245Z INF/device:resources:service 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] STFService up to date
2020-01-01T22:30:09.246Z INF/device:plugins:service 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Launching agent
/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/protobufjs/dist/ProtoBuf.js:2641
                    throw Error("Illegal value for "+this.toString(true)+" of type "+this.type.name+": "+val+" ("+msg+")");
                    ^

Error: Illegal value for Message.Field .TouchDownMessage.y of type float: object (required)
    at Field.<anonymous> (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/protobufjs/dist/ProtoBuf.js:2641:27)
    at Field.ProtoBuf.Reflect.FieldPrototype.verifyValue (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/protobufjs/dist/ProtoBuf.js:2645:25)
    at MessagePrototype.set (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/protobufjs/dist/ProtoBuf.js:1799:63)
    at new Message (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/protobufjs/dist/ProtoBuf.js:1728:42)
    at Socket.<anonymous> (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/lib/units/websocket/index.js:454:31)
    at emitTwo (events.js:126:13)
    at Socket.emit (events.js:214:7)
    at /Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/socket.io/lib/socket.js:528:12
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9)
2020-01-01T22:30:09.775Z FTL/cli:local 6946 [*] Child process had an error ExitError: Exit code "1"
    at ChildProcess.<anonymous> (/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/lib/util/procutil.js:49:23)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
2020-01-01T22:30:09.775Z INF/cli:local 6946 [*] Shutting down all child processes
2020-01-01T22:30:09.782Z INF/util:lifecycle 6948 [app001] Winding down for graceful exit
2020-01-01T22:30:09.783Z INF/util:lifecycle 6950 [proc001] Winding down for graceful exit
2020-01-01T22:30:09.783Z INF/util:lifecycle 6949 [dev001] Winding down for graceful exit
2020-01-01T22:30:09.783Z INF/util:lifecycle 6951 [proc002] Winding down for graceful exit
2020-01-01T22:30:09.783Z INF/util:lifecycle 6952 [reaper001] Winding down for graceful exit
2020-01-01T22:30:09.784Z INF/util:lifecycle 6953 [*] Winding down for graceful exit
2020-01-01T22:30:09.785Z INF/util:lifecycle 6954 [*] Winding down for graceful exit
2020-01-01T22:30:09.785Z INF/auth-mock 6954 [*] Waiting for client connections to end
2020-01-01T22:30:09.786Z INF/util:lifecycle 6955 [*] Winding down for graceful exit
2020-01-01T22:30:09.786Z INF/util:lifecycle 6956 [*] Winding down for graceful exit
2020-01-01T22:30:09.787Z INF/provider 6953 [*] Cleaning up device worker "Ingenic-99431717-1fc06803-f91c0000-886019af"
2020-01-01T22:30:09.797Z WRN/db 6950 [proc001] Connection closed
2020-01-01T22:30:09.797Z INF/provider 6953 [*] Gracefully killing device worker "Ingenic-99431717-1fc06803-f91c0000-886019af"
2020-01-01T22:30:09.797Z WRN/db 6951 [proc002] Connection closed
2020-01-01T22:30:09.798Z WRN/db 6952 [reaper001] Connection closed
2020-01-01T22:30:09.798Z INF/util:lifecycle 6976 [Ingenic-99431717-1fc06803-f91c0000-886019af] Winding down for graceful exit
2020-01-01T22:30:09.802Z WRN/db 6955 [*] Connection closed
2020-01-01T22:30:09.802Z WRN/db 6956 [*] Connection closed
2020-01-01T22:30:09.816Z INF/provider 6953 [*] Device worker "Ingenic-99431717-1fc06803-f91c0000-886019af" has retired
```

When STF doesn't crash' it doesn't seem to be able to get the display info for the device:

```2020-01-01T23:16:49.150Z INF/triproxy 7040 [dev001] PUB socket bound on tcp://127.0.0.1:7114
2020-01-01T23:16:49.153Z INF/triproxy 7040 [dev001] DEALER socket bound on tcp://127.0.0.1:7115
2020-01-01T23:16:49.154Z INF/triproxy 7040 [dev001] PULL socket bound on tcp://127.0.0.1:7116
2020-01-01T23:16:49.154Z INF/triproxy 7039 [app001] PUB socket bound on tcp://127.0.0.1:7111
2020-01-01T23:16:49.157Z INF/triproxy 7039 [app001] DEALER socket bound on tcp://127.0.0.1:7112
2020-01-01T23:16:49.158Z INF/triproxy 7039 [app001] PULL socket bound on tcp://127.0.0.1:7113
2020-01-01T23:16:49.383Z INF/reaper 7043 [reaper001] Subscribing to permanent channel "*ALL"
2020-01-01T23:16:49.387Z INF/reaper 7043 [reaper001] Reaping devices with no heartbeat
2020-01-01T23:16:49.399Z INF/poorxy 7052 [*] Listening on port 7100
2020-01-01T23:16:49.412Z INF/db 7042 [*] Connecting to 127.0.0.1:28015
2020-01-01T23:16:49.417Z INF/db 7043 [reaper001] Connecting to 127.0.0.1:28015
2020-01-01T23:16:49.422Z INF/reaper 7043 [reaper001] Receiving input from "tcp://127.0.0.1:7111"
2020-01-01T23:16:49.423Z INF/reaper 7043 [reaper001] Sending output to "tcp://127.0.0.1:7116"
2020-01-01T23:16:49.425Z INF/db 7041 [*] Connecting to 127.0.0.1:28015
2020-01-01T23:16:49.436Z INF/processor 7042 [proc002] App dealer connected to "tcp://127.0.0.1:7112"
2020-01-01T23:16:49.437Z INF/processor 7042 [proc002] Device dealer connected to "tcp://127.0.0.1:7115"
2020-01-01T23:16:49.462Z INF/processor 7041 [proc001] App dealer connected to "tcp://127.0.0.1:7112"
2020-01-01T23:16:49.463Z INF/processor 7041 [proc001] Device dealer connected to "tcp://127.0.0.1:7115"
2020-01-01T23:16:49.563Z INF/provider 7044 [*] Subscribing to permanent channel "HEqM3lp1RNuKwfw84BjPOw=="
2020-01-01T23:16:49.566Z INF/auth-mock 7045 [*] Listening on port 7120
2020-01-01T23:16:49.599Z INF/provider 7044 [*] Sending output to "tcp://127.0.0.1:7116"
2020-01-01T23:16:49.600Z INF/provider 7044 [*] Receiving input from "tcp://127.0.0.1:7114"
2020-01-01T23:16:49.597Z INF/storage:plugins:image 7050 [*] Listening on port 7103
2020-01-01T23:16:49.605Z INF/provider 7044 [*] Tracking devices
2020-01-01T23:16:49.607Z INF/provider 7044 [*] Found device "Ingenic-99431717-1fc06803-f91c0000-886019af" (device)
2020-01-01T23:16:49.638Z INF/storage:plugins:apk 7051 [*] Listening on port 7104
2020-01-01T23:16:49.661Z INF/provider 7044 [*] Registered device "Ingenic-99431717-1fc06803-f91c0000-886019af"
2020-01-01T23:16:49.662Z INF/reaper 7043 [reaper001] Device "Ingenic-99431717-1fc06803-f91c0000-886019af" is present
2020-01-01T23:16:49.706Z INF/storage:temp 7049 [*] Listening on port 7102
2020-01-01T23:16:49.783Z INF/app 7046 [*] Using pre-built resources
2020-01-01T23:16:49.790Z INF/app 7046 [*] Listening on port 7105
2020-01-01T23:16:49.792Z INF/db 7046 [*] Connecting to 127.0.0.1:28015
2020-01-01T23:16:49.858Z INF/websocket 7048 [*] Subscribing to permanent channel "*ALL"
2020-01-01T23:16:49.862Z INF/websocket 7048 [*] Listening on port 7110
2020-01-01T23:16:49.864Z INF/db 7048 [*] Connecting to 127.0.0.1:28015
2020-01-01T23:16:49.867Z INF/websocket 7048 [*] Sending output to "tcp://127.0.0.1:7113"
2020-01-01T23:16:49.867Z INF/websocket 7048 [*] Receiving input from "tcp://127.0.0.1:7111"
2020-01-01T23:16:50.049Z INF/device:support:push 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Sending output to "tcp://127.0.0.1:7116"
2020-01-01T23:16:50.051Z INF/device 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Preparing device
2020-01-01T23:16:50.084Z INF/api 7047 [*] Subscribing to permanent channel "*ALL"
2020-01-01T23:16:50.123Z INF/api 7047 [*] Listening on port 7106
2020-01-01T23:16:50.156Z INF/api 7047 [*] Sending output to "tcp://127.0.0.1:7113"
2020-01-01T23:16:50.157Z INF/api 7047 [*] Receiving input from "tcp://127.0.0.1:7111"
2020-01-01T23:16:50.264Z INF/db 7047 [*] Connecting to 127.0.0.1:28015
2020-01-01T23:16:50.352Z INF/device:support:sub 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Receiving input from "tcp://127.0.0.1:7114"
2020-01-01T23:16:50.353Z INF/device:support:sub 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Subscribing to permanent channel "*ALL"
2020-01-01T23:16:50.381Z INF/device:support:properties 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Loading properties
2020-01-01T23:16:50.444Z INF/device:support:sdk 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Supports SDK 16
2020-01-01T23:16:50.445Z INF/device:support:abi 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Supports ABIs mips, armeabi-v7a
2020-01-01T23:16:50.553Z INF/device:resources:minicap 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Installing "/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/minicap-prebuilt/prebuilt/mips/bin/minicap" as "/data/local/tmp/minicap"
2020-01-01T23:16:50.553Z INF/device:resources:minicap 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Installing "/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/minicap-prebuilt/prebuilt/mips/lib/android-16/minicap.so" as "/data/local/tmp/minicap.so"
2020-01-01T23:16:52.656Z INF/device:resources:service 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Checking whether we need to install STFService
2020-01-01T23:16:53.370Z INF/device:resources:service 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Running version check
2020-01-01T23:16:56.061Z INF/device:resources:service 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] STFService up to date
2020-01-01T23:16:56.064Z INF/device:plugins:service 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Launching agent
2020-01-01T23:16:56.927Z INF/device:plugins:service 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Agent says: "Listening on @stfagent"
2020-01-01T23:16:57.013Z INF/device:plugins:service 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Launching service
2020-01-01T23:16:57.015Z INF/device:plugins:service 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Agent says: "InputClient started"
2020-01-01T23:16:57.960Z INF/device:plugins:display 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Reading display info
2020-01-01T23:16:58.018Z INF/device:plugins:phone 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Fetching phone info
2020-01-01T23:16:58.044Z INF/device:plugins:identity 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Solving identity
2020-01-01T23:16:58.045Z INF/device:plugins:solo 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Subscribing to permanent channel "YFQf90GsnYErhdUeU49I1hjp1bM="
2020-01-01T23:16:58.046Z INF/device:plugins:screen:stream 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Starting WebSocket server on port 7400
2020-01-01T23:16:58.139Z INF/device:resources:minitouch 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Installing "/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/node_modules/minitouch-prebuilt/prebuilt/mips/bin/minitouch" as "/data/local/tmp/minitouch"
2020-01-01T23:16:58.211Z WRN/device:plugins:data 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Unable to find device data { serial: 'Ingenic-99431717-1fc06803-f91c0000-886019af',
  platform: 'Android',
  manufacturer: 'INGENIC',
  operator: null,
  model: 'SmartTab1',
  version: '4.1',
  abi: 'mips',
  sdk: '16',
  product: 'dwin',
  cpuPlatform: 'jz4780',
  openGLESVersion: '2.0',
  display:
   { id: 0,
     width: 720200592,
     height: 0,
     xdpi: 0,
     ydpi: 0,
     size: 0,
     density: 0,
     fps: 0,
     secure: false,
     rotation: 0,
     url: 'ws://localhost:7400' },
  phone: { imei: '{{{DELETED_B4_PASTE}}}', network: 'UNKNOWN' } }
2020-01-01T23:16:58.215Z INF/device:plugins:touch 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Touch origin is top left
2020-01-01T23:16:58.216Z INF/device:plugins:touch 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Requesting touch consumer to start
2020-01-01T23:16:58.216Z INF/device:plugins:touch 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Launching touch service
2020-01-01T23:16:58.222Z INF/device:plugins:touch 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Connecting to minitouch service
2020-01-01T23:16:58.238Z INF/device:plugins:touch 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] minitouch says: "Type A touch device gt9_ts (800x600 with 2 contacts) detected on /dev/input/event0 (score 20692)"
2020-01-01T23:16:58.334Z INF/device:plugins:touch 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Reading minitouch banner
2020-01-01T23:16:58.335Z INF/device:plugins:touch 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] minitouch says: "Connection established"
2020-01-01T23:16:58.339Z INF/device:plugins:vnc 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Starting VNC server on port 7402
2020-01-01T23:16:58.341Z INF/device:plugins:browser 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Loading browser list
2020-01-01T23:16:58.348Z INF/device:plugins:browser 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Updating browser list
2020-01-01T23:16:58.349Z INF/device:plugins:mute 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Will not mute master volume
2020-01-01T23:16:58.459Z INF/device:resources:minirev 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Installing "/Users/lalalala/.nvm/versions/node/v8.17.0/lib/node_modules/stf/vendor/minirev/mips/minirev" as "/data/local/tmp/minirev"
2020-01-01T23:16:58.549Z INF/device:plugins:forward 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Launching reverse port forwarding service
2020-01-01T23:16:58.555Z INF/device:plugins:forward 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Connecting to reverse port forwarding service
2020-01-01T23:16:59.615Z INF/provider 7044 [*] Providing 0 of 1 device(s); waiting for "Ingenic-99431717-1fc06803-f91c0000-886019af"
2020-01-01T23:16:59.761Z INF/device:plugins:connect 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Listening on port 7401
2020-01-01T23:16:59.764Z INF/device 7053 [Ingenic-99431717-1fc06803-f91c0000-886019af] Fully operational
```
What are the more proper procedures for trying to get a weird device like this to work?

Do I need to add an entry to the device database somewhere?

```
$ stf doctor
+ stf doctor
2020-01-01T23:25:26.714Z INF/cli:doctor 7058 [*] OS Arch: x64
2020-01-01T23:25:26.716Z INF/cli:doctor 7058 [*] OS Platform: darwin
2020-01-01T23:25:26.716Z INF/cli:doctor 7058 [*] OS Platform: 16.7.0
2020-01-01T23:25:26.716Z INF/cli:doctor 7058 [*] Using Node 8.17.0
2020-01-01T23:25:26.729Z INF/cli:doctor 7058 [*] Using ZeroMQ 4.3.2
2020-01-01T23:25:26.742Z INF/cli:doctor 7058 [*] Using GraphicsMagick 1.3.33
2020-01-01T23:25:26.743Z ERR/cli:doctor 7058 [*] ProtoBuf version cannot be detected
2020-01-01T23:25:26.743Z ERR/cli:doctor 7058 [*] ADB version cannot be detected
2020-01-01T23:25:26.748Z INF/cli:doctor 7058 [*] Using RethinkDB 2.3.6
```

**Comment by jachstet-sea** [comment]
Hi, I'm sorry not to be able to help you much, I'll just try to answer some of your questions.
First: MIPS 🎉 . Yay, I really laughed when I read it. Is this the NOVO7 tablet?

> What are the more proper procedures for trying to get a weird device like this to work?

I'd say it's not so much the CPU arch but more the very old version of Android that makes it very hard to use with STF. We have an ARM-based Android 4.2.2 device here and it barely works (screen cap has severe corruption most of the time). Which version of Android is your device running? Do you have any chance to update it?

> Do I need to add an entry to the device database somewhere?

Nope. Unknown devices might show a bit less of useful information on the UI but should work nevertheless

**Comment by mbarnes-arrdude** [comment]
I appreciate any help or pointers you might give me. I am evaluating STF for use in an in-house initiative for remote maintenance for kiosk's MCU UI adapter. So far with my high level testing, with a VPN setup, the system seems very solid with our newer displays, well within our risk tolerance for reliability. I must applaud you on your excellent project and am looking forward to diving deeper and working with it.

The problem I have is that a bunch of our units are this model... I was not involved in the decision... please be kind to me LOL

These are bare board devices for embedded displays and they are not branded. The exact model is hard to tell. It is likely a NOVO7 clone. It has the Ingenics MIPS-alike processor and similar device information to the Karbonn clone of the NOVO7. It is 4.1 as you saw.

I have not found a OS image for it above 4.1. I don't know if you know of anyone that has hacked one. It would be valuable to me for a lot of reasons to find a Lollipop or higher build (even the v4.2.2 Jellybean with the symptoms you describe might be workable), but I do not think that the toolchains released > 4.4 support this processor and I found no OTAs.
